File: [2] Calculations/DataValidation.cs
using System;
namespace CSharp._2__Calculations
{
    public static class DataValidation
    {
        public static dynamic ValidatedNumber(string prompt, short? _min = null, short? _max = null, string _sentinel = null)
        {
            while (true)
            {
                string user_input;
                if (_sentinel != null)
                {
                    Console.Write(prompt.Remove(prompt.Length - 2) + $" (or type '{_sentinel}' to end): ");
                    user_input = Console.ReadLine();
                    if (user_input.Equals(_sentinel, StringComparison.CurrentCultureIgnoreCase))
                        return null;
                }
                else
                {
                    Console.Write(prompt);
                    user_input = Console.ReadLine();
                }

                try
                {
                    float number = Convert.ToSingle(user_input);
                    if ((_min == null) || (_min <= number) && (_max == null) || (_max >= number))
                        return number;
                    else
                        Console.WriteLine($"Input out of range. [{_min}-{_max}]");
                }
                catch (FormatException)
                {
                    Console.WriteLine("! Error: Input is NaN");
                }
            }
        }
    }
}

File: Python/DataValidation.py
def ValidatedNumber(prompt, _min=None, _max=None, _sentinel=None):
    while True:
        if _sentinel is not None:
            user_input = input(prompt[:-2] + f" (or type \'{_sentinel}\' to end): ")
            if user_input.upper() == _sentinel:
                return None
        else:
            user_input = input(prompt)
        try:
            number = float(user_input)
            if (_min is None or _min <= number) and (_max is None or _max >= number):
                return number
            else:
                print(f"Input out of range. [{_min}-{_max}]")
        except OverflowError:
            print("! Error: Input is too large.")
        except ValueError:
            print("! Error: Input is Not-a-Number.")
